static/app/components/replays/replayPlayer.tsx

```diff
@@ -0,0 +1,95 @@
+import {useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Consumer as ReplayContextProvider} from 'sentry/components/replays/replayContext';
+
+interface Props {
+  className?: string;
+}
+
+type Dimensions = {height: number; width: number};
+type RootElem = null | HTMLDivElement;
+
+type RootProps = {
+  initRoot: (root: RootElem) => void;
+  videoDimensions: Dimensions;
+  className?: string;
+};
+
+function PlayerRoot({className, initRoot, videoDimensions}: RootProps) {
+  const windowEl = useRef<HTMLDivElement>(null);
+  const viewEl = useRef<HTMLDivElement>(null);
+
+  const [windowDimensions, setWindowDimensions] = useState<Dimensions>();
+
+  // Create the `rrweb` instance which creates an iframe inside `viewEl`
+  useEffect(() => initRoot(viewEl.current), [viewEl.current]);
+
+  // Read the initial width/height where the player will be inserted, to setup view transform/scaling
+  // Also listen for resize events and so we can update view transform/scaling
+  useEffect(() => {
+    const handleResize = () =>
+      setWindowDimensions(windowEl.current?.getBoundingClientRect());
+    handleResize();
+    window.addEventListener('resize', handleResize);
```

Comment:
Should this be throttled/debounced? or should we use resize observer?

Reply:
Resize observer could be cool, i didn't know about that before!

I didn't measure the impact of adding throttle or debounce, I can follow up on this diff with that I think.

Reply:
don't need to follow up.

I found `useResizeObserver` which is cool, and also replaced `getBoundingClientRect` with reads to `.clientWidth` and `.clientHeight`, so the browser doesn't need to calculate all the values.